task_3/theater-no-orm/pkg/data/theater.go

```diff
@@ -0,0 +1,427 @@
+	rows, err := u.db.Query(readAllTicketsQuery)
+	if err != nil {
+		return nil, fmt.Errorf("can't get tickets from database, error:%w", err)
+	}
+	for rows.Next() {
+		var temp Ticket
+		err = rows.Scan(&temp.Id, &temp.PerformanceName, &temp.GenreName, &temp.PerformanceDuration,
+			&temp.DateTime, &temp.HallName, &temp.HallCapacity, &temp.LocationAddress, &temp.LocationPhoneNumber,
+			&temp.SectorName, &temp.Place, &temp.Price, &temp.DateOfIssue, &temp.Paid, &temp.Reservation, &temp.Destroyed)
+		if err != nil {
+			return nil, fmt.Errorf("can't scan tickets from database, error:%w", err)
+		}
+		tickets = append(tickets, temp)
+	}
+	return tickets, nil
+}
+
+func (u TheaterData) ReadAllPosters() ([]Poster, error) {
+	var posters []Poster
+	rows, err := u.db.Query(readAllPostersQuery)
+	if err != nil {
+		return nil, fmt.Errorf("can't get posters from database, error:%w", err)
+	}
+	for rows.Next() {
+		var temp Poster
+		err = rows.Scan(&temp.Id, &temp.PerformanceName, &temp.GenreName, &temp.PerformanceDuration,
+			&temp.DateTime, &temp.HallName, &temp.HallCapacity, &temp.LocationAddress, &temp.LocationPhoneNumber,
+			&temp.Comment)
+		if err != nil {
+			return nil, fmt.Errorf("can't scan posters from database, error:%w", err)
+		}
+		posters = append(posters, temp)
+	}
+	return posters, nil
+}
+
+func (u TheaterData) ReadAllUsers(account Account) ([]User, error) {
+	var users []User
+	rows, err := u.db.Query(readAllUsersQuery, account.Id)
+	if err != nil {
+		return nil, fmt.Errorf("can't get users from database, error:%w", err)
+	}
+	for rows.Next() {
+		var temp User
+		err = rows.Scan(&temp.Id, &temp.FirstName, &temp.LastName, &temp.Role,
+			&temp.LocationAddress, &temp.LocationPhoneNumber, &temp.PhoneNumber)
+		if err != nil {
+			return nil, fmt.Errorf("can't scan users from database, error:%w", err)
+		}
+		users = append(users, temp)
+	}
+	return users, nil
+}
+
+func (u TheaterData) AddAccount(account Account) error {
+	_, err := u.db.Exec(insertAccount, account.FirstName, account.LastName, account.PhoneNumber, account.Email)
+	if err != nil {
+		return fmt.Errorf("can't inser account to database, error: %w", err)
+	}
+	return nil
+}
+
+func (u TheaterData) AddGenre(genre Genre) error {
+	_, err := u.db.Exec(insertGenre, genre.Name)
+	if err != nil {
+		return fmt.Errorf("can't inser genre to database, error: %w", err)
+	}
+	return nil
+}
+
+func (u TheaterData) AddHall(hall Hall) error {
+	_, err := u.db.Exec(insertHall, hall.AccountId, hall.Name, hall.Capacity, hall.LocationId)
+	if err != nil {
+		return fmt.Errorf("can't inser hall to database, error: %w", err)
+	}
+	return nil
+}
+
+func (u TheaterData) AddLocation(location Location) error {
+	_, err := u.db.Exec(insertLocation, location.AccountId, location.Address, location.PhoneNumber)
+	if err != nil {
+		return fmt.Errorf("can't inser location to database, error: %w", err)
+	}
+	return nil
+}
+
+func (u TheaterData) AddPerformance(performance Performance) error {
```

Comment:
could we also return the `id` of inserted perfomance

Reply:
Done